Dockerfile

```diff
@@ -1,33 +1,31 @@
-FROM golang:1.9.2-alpine3.7
+FROM golang:1
 
 ENV PROJECT=mongo-hot-backup
-COPY . /${PROJECT}-sources/
 
-ADD . /source/
-RUN apk --no-cache --virtual .build-dependencies add --update bash git \
-  && ORG_PATH="github.com/Financial-Times" \
-  && REPO_PATH="${ORG_PATH}/${PROJECT}" \
-  && mkdir -p $GOPATH/src/${ORG_PATH} \
-  # Linking the project sources in the GOPATH folder
-  && ln -s /${PROJECT}-sources $GOPATH/src/${REPO_PATH} \
-  && cd $GOPATH/src/${REPO_PATH} \
-  && BUILDINFO_PACKAGE="${ORG_PATH}/${PROJECT}/vendor/${ORG_PATH}/service-status-go/buildinfo." \
-  && VERSION="version=$(git describe --tag --always 2> /dev/null)" \
-  && DATETIME="dateTime=$(date -u +%Y%m%d%H%M%S)" \
-  && REPOSITORY="repository=$(git config --get remote.origin.url)" \
-  && REVISION="revision=$(git rev-parse HEAD)" \
-  && BUILDER="builder=$(go version)" \
-  && LDFLAGS="-X '"${BUILDINFO_PACKAGE}$VERSION"' -X '"${BUILDINFO_PACKAGE}$DATETIME"' -X '"${BUILDINFO_PACKAGE}$REPOSITORY"' -X '"${BUILDINFO_PACKAGE}$REVISION"' -X '"${BUILDINFO_PACKAGE}$BUILDER"'" \
-  && echo "Build flags: $LDFLAGS" \
-  && echo "Fetching dependencies..." \
-  && go get -u github.com/kardianos/govendor \
-  && $GOPATH/bin/govendor sync \
-  && go build -ldflags="${LDFLAGS}" \
-  && mv ${PROJECT} /${PROJECT} \
-  && apk del .build-dependencies \
-  && rm -rf $GOPATH /var/cache/apk/*
+ENV ORG_PATH="github.com/Financial-Times"
+ENV SRC_FOLDER="${GOPATH}/src/${ORG_PATH}/${PROJECT}"
+ENV BUILDINFO_PACKAGE="${ORG_PATH}/service-status-go/buildinfo."
 
+COPY . ${SRC_FOLDER}
+WORKDIR ${SRC_FOLDER}
+
+# Install dependancies and build app
+RUN VERSION="version=$(git describe --tag --always 2> /dev/null)" \
+    && DATETIME="dateTime=$(date -u +%Y%m%d%H%M%S)" \
+    && REPOSITORY="repository=$(git config --get remote.origin.url)" \
+    && REVISION="revision=$(git rev-parse HEAD)" \
+    && BUILDER="builder=$(go version)" \
+    && LDFLAGS="-X '"${BUILDINFO_PACKAGE}$VERSION"' -X '"${BUILDINFO_PACKAGE}$DATETIME"' -X '"${BUILDINFO_PACKAGE}$REPOSITORY"' -X '"${BUILDINFO_PACKAGE}$REVISION"' -X '"${BUILDINFO_PACKAGE}$BUILDER"'" \
+    && CGO_ENABLED=0 go build -mod=readonly -a -o /artifacts/${PROJECT} -ldflags="${LDFLAGS}" \
+    && echo "Build flags: ${LDFLAGS}"
+
+EXPOSE 8080
```

Comment:
I am not sure why and if this `EXPOSE 8080` is necessary, I don't see it in the Dockerfile in master.
@mchompalova or someone else who is familiar, can you please elaborate on that change?

Reply:
Those are mostly used as a documentation i.e. hinting which ports will be exposed when the container is run. I think it is fine if we drop it since we don't use `EXPOSE` in the majority of our Dockerfiles.